lib/hex_web/owners.ex

```diff
@@ -0,0 +1,67 @@
+defmodule HexWeb.Owners do
+  import Ecto
+  alias HexWeb.{User, Package, Repo}
+
+  def all(package) do
+    assoc(package, :owners)
+    |> Repo.all
+  end
+
+  def find(email) do
+    Repo.get_by!(User, email: email)
+  end
+
+  def add(package, owner, opts \\ []) do
+    multi =
+      Ecto.Multi.new
+      |> Ecto.Multi.insert(:owner, Package.build_owner(package, owner))
+      |> audit(opts)
+
+    case HexWeb.Repo.transaction(multi) do
+      {:ok, _} ->
+        owners = all(package)
+
+        HexWeb.Mailer.send(
+          "owner_add.html",
+          "Hex.pm - Owner added",
+          Enum.map(owners, & &1.email),
+          username: owner.username,
+          email: owner.email,
+          package: package.name
+        )
+        :ok
+
+      {:error, :owner, changeset, _} ->
+        {:error, changeset}
+    end
+  end
+
+  def remove(package, owner, opts \\ []) do
+    owners = all(package)
+
+    if length(owners) == 1 do
+      {:error, :last_owner}
+    else
+      multi =
+        Ecto.Multi.new
+        |> Ecto.Multi.delete_all(:package_owner, Package.owner(package, owner))
+        |> audit(opts)
+
+      {:ok, _} = HexWeb.Repo.transaction(multi)
+
+      HexWeb.Mailer.send(
+        "owner_remove.html",
+        "Hex.pm - Owner removed",
+        Enum.map(owners, & &1.email),
+        username: owner.username,
+        email: owner.email,
+        package: package.name
+      )
+      :ok
+    end
+  end
+
+  defp audit(multi, opts) do
+    Keyword.fetch!(opts, :audit_fun).(multi)
+  end
```

Comment:
I liked it a lot too. But since we are separating the controllers and the crud now I think we should go all the way and remove the audit crud from the controllers also :)